astronomy-pipeline/test_astronomy_transform.py

```diff
@@ -0,0 +1,318 @@
+        mock_load.return_value = {}
+
+        load_from_file('test')
+
+        assert mock_load.called
+        assert mock_open.called
+
+    @patch('json.load')
+    @patch('builtins.open')
+    def test_load_from_file_uses_correct_calls(self, mock_open, mock_load):
+        '''Tests that the open method is called with the correct
+        values.'''
+        mock_open.return_value.__enter__.return_value = MagicMock()
+        mock_load.return_value = {}
+
+        load_from_file('test')
+
+        mock_open.assert_called_once_with('test', 'r', encoding='UTF-8')
+
+    @patch('json.load')
+    @patch('builtins.open')
+    def test_load_from_file_returns_dict(self, mock_open, mock_load):
+        '''Tests that a dictionary is returned from the load from file function.'''
+        mock_open.return_value.__enter__.return_value = MagicMock()
+        mock_load.return_value = {}
+
+        result = load_from_file('test')
+
+        assert isinstance(result, dict)
+
+
+class TestGetDataIntoDataFrame():
+    '''Tests for the get data into dataframe function.'''
+
+    @patch('pandas.concat')
+    @patch('pandas.DataFrame')
+    def test_get_data_into_dataframe_correct_methods(self, mock_dataframe, mock_concat):
+        '''Tests that the correct pandas methods are called when the function runs.'''
+        get_data_into_dataframe({'body_positions': {1: {1: 'test'}}})
+
+        assert mock_dataframe.called
+        assert mock_concat.called
+
+    def test_get_data_into_dataframe_missing_body_positions(self):
+        """Test get_data_into_dataframe with missing body positions."""
+        with pytest.raises(KeyError):
+            get_data_into_dataframe({})
+
+
+class TestGetBodyMapping():
+    '''Tests for the get body mapping function.'''
+
+    @patch('astronomy_transform.get_db_connection')
+    def test_get_body_mapping_correct_methods(self, mock_get_db_connection):
+        '''Tests that the correct function and methods are 
+        called within the function.'''
+        mock_conn = MagicMock()
+        mock_cursor = MagicMock()
+        mock_get_db_connection.return_value.__enter__.return_value = mock_conn
+        mock_conn.cursor.return_value = mock_cursor
+
+        get_body_mapping()
+
+        assert mock_cursor.execute.called
+        assert mock_get_db_connection.called
+        assert mock_conn.cursor.called
+
+    @patch('astronomy_transform.get_db_connection')
+    def test_get_body_mapping_correct_call(self, mock_get_db_connection):
+        '''Tests that the correct input is used for the execute method.'''
+        mock_conn = MagicMock()
+        mock_cursor = MagicMock()
+        mock_get_db_connection.return_value.__enter__.return_value = mock_conn
+        mock_conn.cursor.return_value = mock_cursor
+
+        get_body_mapping()
+
+        mock_cursor.execute.assert_called_once_with(
+            'SELECT body_id, body_name from body;')
+
+    @patch('astronomy_transform.get_db_connection')
```

Comment:
Good mocking of database connections to avoid interacting with our RDS for testing.